stl/inc/future

```diff
@@ -336,10 +349,11 @@ public:
 
     void _Set_exception_raw(exception_ptr _Exc, unique_lock<mutex>* _Lock,
         bool _At_thread_exit) { // store a result while inside a locked block
-        if (_Has_stored_result) {
+        if (_Already_has_stored_result()) {
             _Throw_future_error(make_error_code(future_errc::promise_already_satisfied));
         }
 
+        _STL_ASSERT((_Exc != nullptr), "promise<T>::set_exception(nullptr)");

```

Comment:
Thanks for adding this precondition check. The assertion message is accurate, but I think it would be helpful to additionally cite the Standardese, beyond mentioning what was called. Adding "is invalid, see N4901 [futures.promise]/20" would be sufficient. (Mostly I'm thinking of users with existing codebases who upgrade and see this assertion emitted - seeing an assertion without an explanation may make them think they've encountered a toolset bug, whereas a message citing the Standardese is more clearly diagnosing an existing problem in their code.)